### PR TITLE
MAINT: add warning to numpy.distutils for LDFLAGS append behavior.

### PR DESCRIPTION
**Pull request by rgommers**
See gh-7427.

@isuruf given that this is a tweak of your PR, would you mind looking this over?

### CODE REVIEW

File: numpy/distutils/fcompiler/environment.py [hunk 1]
```diff
@@ -57,6 +58,12 @@ def _get_var(self, name, conf_desc):
                 if var and append and os.environ.get('NPY_DISTUTILS_APPEND_FLAGS', '0') == '1':
                     var = var + [envvar_contents]
                 else:
+                    if envvar in ['LDFLAGS', 'LDSHARED']:
+                        msg = "{} is used as is, not appended ".format(envvar) + \
+                              "to flags already defined " + \
+                              "by numpy.distutils! Use NPY_DISTUTILS_APPEND_FLAGS=1 " + \
+                              "to obtain appending behavior instead"
```
Comment (isuruf) [inline numpy/distutils/fcompiler/environment.py]: Maybe warn about `NPY_DISTUTILS_APPEND_FLAGS=1` being default in a future release?

**Comment by charris** [comment]
Pushing this off until the test failure is resolved.

**Comment by mattip** [comment]
The failing test, `test_fcompiler_flags`, is getting tripped up on the new warning. It needs a warning-filter context manager.

File: numpy/distutils/fcompiler/environment.py [hunk 2]
```diff
@@ -57,6 +58,12 @@ def _get_var(self, name, conf_desc):
                 if var and append and os.environ.get('NPY_DISTUTILS_APPEND_FLAGS', '0') == '1':
                     var = var + [envvar_contents]
                 else:
+                    if envvar in ['LDFLAGS', 'LDSHARED']:
```
[the remark this answers is not in the document]
Comment (rgommers) [reply numpy/distutils/fcompiler/environment.py]: thanks, makes sense. I started with all but ran into too many warnings; with `if var and append` that's not an issue though.

File: numpy/distutils/fcompiler/environment.py (hunk 1 above)
Comment (rgommers) [reply numpy/distutils/fcompiler/environment.py]: good point, done

**Comment by rgommers** [comment]
All green finally.

**Pull request merged**